### DISCUSSION TITLE
authenticated server to external server requests

### DISCUSSION THREAD

**Discussion by clyncha** [vercel/next.js#60730] [Help]
### Summary

How can I go about making authenticated server to external server requests with the new architecture of react server side components?

My team is using Auth0 for authentication and we require the bearer token on requests sent from the app. I have an issue where I cannot use getAccessToken from Auth0 because it attempts to update cookies.

I tried the middleware route but that doesn't unblock me. The main issue is that I need to 
1) get the token to pass
2) persist changes to the refresh token

**Comment by kzunino** [comment]
The issue of having React Server Components make authenticated requests to external services doesn't seem to be addressed in most (if not all) of the examples I've seen online. We have to deduce this realization from the fact that even though we can make API calls in RSCs, we cannot write cookies in RSCs. This limits the ability for us to make _authenticated_ requests to external services—at least from what I can gather.
